Update and add fields for new registration API

mkangia opened the pull request:
[Changes](https://docs.google.com/spreadsheets/d/1LK6xQHldOT8tf3ctcypAyUlIuTN-C25zhyIC1jHJeS0/edit#gid=917275059)

mkangia committed:
Update and add fields for new registration API

mkangia committed:
fix lint

mkangia committed:
retain legacy payload generator and add new v2 repeater and payload generator to have easy transition and removal of legacy code later

mkangia committed:
lint [skip ci]

proteusvacuum commented on custom/enikshay/integrations/nikshay/repeater_generator.py:
```diff
@@ -93,8 +94,10 @@ def _base_properties(self, repeat_record):
+            "source": ENIKSHAY_ID,
             "IP_From": server_ip,
             "IP_FROM": server_ip,
+            "ip_address": server_ip,
```
lol

proteusvacuum reviewed (commented):
looking like a good start! no major comments.

mkangia committed:
rename

mkangia committed:
inherit from Base class only since overwriting everything anyway.
fix other errors

mkangia committed:
Add mapping for not_known and use that when occupation not available

mkangia committed:
Align notifications for two different APIs

Trigger Registration, HIVTest, FollowUP and Treatment Outcome only for patients diagnosed before a specific date using legacy API and otherwise use new one

mkangia committed:
rename to better name

mkangia committed:
use diagnosing facility id for v2 since notifications are on diagnosis

mkangia committed:
add phi name

mkangia committed:
identify cases notified via v2

mkangia committed:
lint

mkangia committed:
add phi_name to the tuple missed earlier

mkangia committed:
add key population for v2 api

mkangia commented on custom/enikshay/integrations/nikshay/field_mappings.py:
```diff
@@ -156,3 +157,25 @@
+    'unmarried': 'S',
+    'married': 'M',
+    'widowed': 'W',
+    'separated': 'Se'
```
W and Se should just become 'other'.

mkangia committed:
Add mapping for two new fields under occupation in new ToG [skip ci]

mkangia committed:
add mapping for area

mkangia committed:
update mappings for area and marital status

mkangia committed:
take the first option in case of multiple selection

mkangia committed:
undo changes for earlier version of v2 in payload generator

mkangia committed:
add fields according to new v2 api

mkangia committed:
remove phi_name as its not needed anymore

mkangia committed:
fix field names with extra space

mkangia committed:
fix error with parsing date, upper case economic status, add codes for state and district

mkangia committed:
return the code

mkangia committed:
get district code from district_codes

mkangia committed:
update field names

mkangia committed:
fix marital status mappings

proteusvacuum commented:
let me know when you want me to review this again

mkangia committed:
add not available fields

mkangia commented:
sure @proteusvacuum 

this is not intended to be available as a repeater anymore i.e this does not need to check on updates on cases and notify. We would just get a list of episode ids and then just notify them using the payload generator. Would remove the repeater and other related code. Any inputs/remarks?

mkangia committed:
update key names

mkangia committed:
remove redundant changes and adapt to things being removed

mkangia committed:
Resolve conflicts: Merge master

mkangia committed:
add dummy values for missing fields

mkangia committed:
handle case where the value is present but blank/empty string

mkangia committed:
update dummy values for fields

mkangia committed:
send mappings for multiple occupations

mkangia committed:
revert changes to share location ids, just send nikshay codes

mkangia committed:
use const instead

mkangia committed:
Remove redundant file added by mistake

mkangia committed:
fix issue when returning property_value, break address into three fields, blank fields

mkangia committed:
update to use hiv status as other instead of unknown

mkangia committed:
send NA instead of unknown for socioeconomic status and send the dummy number instead of 0

mkangia committed:
send null in case of town missing and limit landmark to 50 chars

mkangia committed:
send dummy phone number when number not correct or missing, truncate to 50, and send state and district code itself when needed

mkangia committed:
revert changes for hiv status as other

proteusvacuum commented:
:cry:
